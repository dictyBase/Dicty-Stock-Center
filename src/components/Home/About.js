// @flow
import React, { Component } from "react"
import { connect } from "react-redux"
import Grid from "@material-ui/core/Grid"
import Skeleton, { SkeletonTheme } from "react-loading-skeleton"
import InlineEditor from "components/InlineEditor"
import { fetchInfoPage } from "actions/page"

const slugName = "dsc-about"

type Props = {
  /** the Auth object taken from the current state */
  auth: Object,
  /** Action creator that fetches data from API */
  fetchInfoPage: Function,
  /** the Page object taken from the current state */
  page: Object,
  /** Checks if data is currently being fetched */
  isFetching: boolean,
}

/**
 * About fetches and displays the About page content.
 */

class About extends Component<Props> {
  static defaultProps = {
    page: {
      data: {
        attributes: {},
      },
    },
  }
  componentDidMount() {
    this.props.fetchInfoPage(slugName)
  }
  render() {
    const { isFetching, page } = this.props

    if (!isFetching && page.data.attributes.content) {
      return (
        <InlineEditor
          auth={this.props.auth}
          page={this.props.page}
          slug={slugName}
        />
      )
    }
    return (
      <Grid container justify="center">
        <Grid item xs={12}>
          <SkeletonTheme color="#D3D3D3	" highlightColor="#DCDCDC">
            <Skeleton count={7} />
            <br />
            <br />
            <Skeleton count={7} />
          </SkeletonTheme>
        </Grid>
      </Grid>
    )
  }
}

const mapStateToProps = state => ({
  auth: state.auth,
  isFetching: state.page.isFetching,
  page: state.page[slugName],
})

export { About }
export default connect<*, *, *, *, *, *>(mapStateToProps, { fetchInfoPage })(
  About,
)
