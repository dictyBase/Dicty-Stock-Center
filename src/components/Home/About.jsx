// @flow
import React, { Component } from 'react'
import { connect } from 'react-redux'
import Skeleton from 'react-loading-skeleton'
import AboutInlineEditor from '../editor/AboutInlineEditor'
import { fetchInfoPage } from 'actions/page'
import { Flex, Box } from 'rebass'

type Props = {
  auth: Object,
  fetchInfoPage: Function,
  page: Object,
  isFetching: boolean
}

class About extends Component<Props> {
  displayName = 'homepage about us component'
  // set defaultprops to prevent console warnings
  static defaultProps = {
    page: {
      data: {
        attributes: {}
      }
    }
  }
  componentDidMount() {
    this.props.fetchInfoPage('dsc-about')
  }
  render() {
    const { isFetching, page } = this.props

    if (!isFetching && page.data.attributes.content) {
      return <AboutInlineEditor auth={this.props.auth} page={this.props.page} />
    }
    return (
      <Flex justify="center">
        <Box w={'95%'}>
          <Skeleton count={5} />
          <br /><br />
          <Skeleton count={5} />
        </Box>
      </Flex>
    )
  }
}

const mapStateToProps = state => {
  const slugName = 'dsc-about'
  return {
    auth: state.auth,
    isFetching: state.page.isFetching,
    page: state.page[slugName]
  }
}

export default connect(mapStateToProps, { fetchInfoPage })(About)
