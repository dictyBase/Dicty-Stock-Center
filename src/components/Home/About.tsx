import React from "react"
import { useQuery } from "@apollo/react-hooks"
import Grid from "@material-ui/core/Grid"
import Skeleton, { SkeletonTheme } from "react-loading-skeleton"
import InlineEditor from "components/InlineEditor"
import { GET_CONTENT_BY_SLUG } from "queries/queries"

const slugName = "dsc-about"

/**
 * About fetches and displays the About page content.
 */

const About = () => {
  const { loading, error, data } = useQuery(GET_CONTENT_BY_SLUG, {
    variables: {
      slug: slugName,
    },
  })

  if (loading) {
    return (
      <Grid container justify="center">
        <Grid item xs={12}>
          <SkeletonTheme color="#D3D3D3	" highlightColor="#DCDCDC">
            <Skeleton count={7} />
            <br />
            <br />
            <Skeleton count={7} />
          </SkeletonTheme>
        </Grid>
      </Grid>
    )
  }

  if (error) {
    return <div>got an error</div>
  }

  return <InlineEditor data={data.contentBySlug} />
}

export default About
