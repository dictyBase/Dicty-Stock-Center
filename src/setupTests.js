import { configure } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import "jest-enzyme"

global.console = {
  log: jest.fn(),
  error: jest.fn(),
  warn: jest.fn(),
}

configure({ adapter: new Adapter() })
