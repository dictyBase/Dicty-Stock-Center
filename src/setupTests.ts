import { configure } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import "jest-enzyme"
import { sprintf } from "sprintf-js"

let globalAny = global as any

globalAny.console = {
  log: (msg: string, args: any) => {
    if (typeof msg !== "string") {
      return
    }
    const str = sprintf(msg, args)
    process.stderr.write(str + "\n")
  },
  error: jest.fn(),
  warn: jest.fn(),
}

configure({ adapter: new Adapter() })
