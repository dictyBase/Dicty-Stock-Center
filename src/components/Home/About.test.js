import React from "react"
import renderer from "react-test-renderer"
import { shallow, mount } from "enzyme"
import sinon from "sinon"
import "jest-styled-components"
import "../../setupTests"
import { About } from "./About"

test("matching a snapshot of About", () => {
  const fetchInfoPage = () => {}
  const page = {
    data: {
      attributes: {},
    },
  }

  const component = renderer.create(
    <About page={page} fetchInfoPage={fetchInfoPage} isFetching={false} />,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

describe("Home/About", () => {
  const props = {
    page: {
      data: {
        attributes: {},
      },
    },
    fetchInfoPage: () => {},
    isFetching: false,
  }
  const slugName = "dsc-about"

  it("calls componentDidMount", () => {
    sinon.spy(About.prototype, "componentDidMount")
    const wrapper = mount(<About {...props} />)
    expect(About.prototype.componentDidMount.calledOnce).toEqual(true)
  })
})
