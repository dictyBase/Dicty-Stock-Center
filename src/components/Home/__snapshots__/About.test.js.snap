// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of About 1`] = `
<div
  className="sc-htpNat iZBBQa sc-bdVaJa iHZvIS"
  is={null}
>
  <div
    className="sc-bdVaJa FHiEA"
    is={null}
  >
    <div
      className="react-loading-theme-0"
    >
      <span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
      </span>
      <br />
      <br />
      <span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
      </span>
    </div>
  </div>
</div>
`;
